ReactiveCocoa/Swift/EventLogger.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  EventLogger.swift
+//  ReactiveCocoa
+//
+//  Created by Rui Peres on 30/04/2016.
+//  Copyright © 2016 GitHub. All rights reserved.
+//
+
+import Foundation
+
+public enum LoggingEvent: String {
+	case Started, Next, Completed, Failed, Terminated, Disposed, Interrupted
+	
+	public static let allEvents: Set<LoggingEvent> = {
+		return [.Started, .Next, .Completed, .Failed, .Terminated, .Disposed, .Interrupted]
+	}()
```

Comment:
Gonna restart the discussion about separate `LoggingEvent` `enum`s for `Signal` and `SignalProducer` since the previous one was collapsed.

You mentioned repeated code. The cases would be repeated, and `SignalProducer.logEvents` wouldn't be able to `lift, but I don't think it'd be much more code.

I'm concerned that people will expected `Started` to work on `Signal`s when it doesn't. Separate types would let use enforce correctness with the types.

What do you think? (I'm not trying to force you; just trying to have a discussion. :relaxed:)